lib/pbench/server/cache_manager.py

```diff
@@ -424,12 +424,14 @@ def create(cls, tarball: Path, controller: "Controller") -> "Tarball":
             # log it but do not abort
             controller.logger.error("Unable to set SELINUX context for {}: {}", name, e)
 
-        # If we were able to copy both files, remove the originals
+        # If we were able to copy both files, remove the originals. If we moved
+        # the files above, instead of copying them, these will no longer exist
+        # and we'll ignore that condition silently.
         try:
-            tarball.unlink()
-            md5_source.unlink()
+            tarball.unlink(missing_ok=True)
+            md5_source.unlink(missing_ok=True)
         except Exception as e:
-            controller.logger.error("Error removing incoming dataset {}: {}", name, e)
+            controller.logger.error("Error removing staged upload {}: {}", name, e)
```

Comment:
If the `unlink()` for `tarball` fails, we'll leave behind the `md5_source`...do we care?  (I mean, if we can't delete the one, we probably can't delete the other....)  Otherwise, the two `unlink()` calls should presumably be in separate `try` blocks.

Reply:
I wasn't inclined to care at the time. Is it really worth complicating the code? Sigh.

Reply:
If you're going to do another round anyway, I would say, "yes".